src/rgw/rgw_op.cc

```diff
@@ -6514,6 +6514,22 @@ void RGWDeleteMultiObj::execute()
 
     send_partial_response(*iter, del_op.result.delete_marker,
 			  del_op.result.version_id, op_ret);
+
+    const auto obj_state = obj_ctx->get_state(obj);
+    s->object = obj.key;
```

Comment:
aren't we in a range-for over 1..m objects here?  I haven't read older versions of this code that I remember, but it seems I'm surprised we would assigned req_state's object or size members here.  I don't see where in the current PR this is really doing anything (other than letting some state leak out of this routine)?

Reply:
agree it is a bit of a hack, but the `publish` call already expect req_state as a parameter.
other option would be:
- add key and size as explicit parameters to `publish`
- create a local copy of req_state and set it with key and size

Reply:
@mattbenjamin done, added name and size as parameters to `publish`